### PR TITLE
Uses cuda pypi nccl

### PR DESCRIPTION
**Pull request by syed-ahmed**
This PR adds NCCL from pypi as a dependency. See https://github.com/pytorch/builder/pull/1196 for testing plan.

### CODE REVIEW

File: .github/workflows/generated-linux-binary-manywheel-nightly.yml [hunk 1]
```diff
@@ -25,6 +25,10 @@ env:
   BINARY_ENV_FILE: /tmp/env
   BUILD_ENVIRONMENT: linux-binary-manywheel
   BUILDER_ROOT: /builder
+  CU117_DEPENDENCIES: "nvidia-cuda-runtime-cu11==11.7.99;\
```
Comment (atalman) [inline .github/workflows/generated-linux-binary-manywheel-nightly.yml:28]: Was this file manually edited ? This is generated file, should not be edited manually

Comment (atalman) [reply .github/workflows/generated-linux-binary-manywheel-nightly.yml:28]: Also please keep the previous logic using PYTORCH_EXTRA_INSTALL_REQUIREMENTS

File: .github/scripts/generate_binary_build_matrix.py [hunk 2]
```diff
@@ -219,9 +219,10 @@ def generate_wheels_matrix(os: str,
                         "container_image": WHEEL_CONTAINER_IMAGES[arch_version],
                         "package_type": package_type,
                         "pytorch_extra_install_requirements":
-                        "nvidia-cuda-runtime-cu11;"
+                        "nvidia-cuda-runtime-cu11==11.7.99;"
                         "nvidia-cudnn-cu11==8.5.0.96;"
-                        "nvidia-cublas-cu11==11.10.3.66",
+                        "nvidia-cublas-cu11==11.10.3.66;"
+                        "nvidia-nccl-cu11==2.14.3",
                         "build_name":
```
Comment (atalman) [inline .github/scripts/generate_binary_build_matrix.py:226]: Should we also add libnvrtc here ? as in this PR https://github.com/pytorch/builder/pull/1183 
And remove libnvrtc from patch ?

**Review by atalman — CHANGES_REQUESTED** [review]
Thank you Syed! Please see comments below.

**Comment by syed-ahmed** [comment]
Merged this PR here: https://github.com/pytorch/pytorch/pull/89944. Closing.

**Pull request closed without merging**